lms/djangoapps/django_comment_client/utils.py

```diff
@@ -981,6 +1033,7 @@ def get_group_name(group_id, course_discussion_settings):
 
     Returns: learner-facing name of the Group, or None if no such group exists
     """
+    # TODO: found it
     group_names_by_id = get_group_names_by_id(course_discussion_settings)
     return group_names_by_id[group_id] if group_id in group_names_by_id else None
```

Comment:
This is the big one; we're getting *all* group names to get the data for a single group, then promptly throw it away and repeat on the next check